src/nyancat.c

```diff
@@ -485,7 +484,7 @@ int main(int argc, char ** argv) {
 								/* This was a response to the TTYPE command, meaning
 								 * that this should be a terminal type */
 								alarm(2);
-								strcpy(term, (char *)&sb[2]);
+								term = (char *)&sb[2];
```

Comment:
Makes sense, thanks for pointing this out. I modified my diff to use `strndup`, which should fix the issue.